apps/meteor/client/hooks/useUserDisplayName.ts

```diff
@@ -1,10 +1,13 @@
 import type { IUser } from '@rocket.chat/core-typings';
-import { useSetting } from '@rocket.chat/ui-contexts';
+import { useUserPreference, useSetting } from '@rocket.chat/ui-contexts';
 
 import { getUserDisplayName } from '../lib/getUserDisplayName';
 
 export const useUserDisplayName = ({ name, username }: Pick<IUser, 'name' | 'username'>): string | undefined => {
-	const useRealName = useSetting('UI_Use_Real_Name');
+	const messagesLayoutPreference = useUserPreference('messagesLayout');
+	const defaultMessagesLayout = useSetting('Accounts_Default_User_Preferences_messagesLayout');
+	const useRealName =
+		messagesLayoutPreference !== 'default' ? messagesLayoutPreference !== 'username' : defaultMessagesLayout !== 'username';
```

Comment:
This one is also interesting to move into a helper function, it's the same code being used 9 times along the app.

_note: Might need to think on a different name for this function, or rename the other I suggested_
```
export const useRealName = () => {
	const messagesLayoutPreference = useUserPreference('messagesLayout');
	const defaultMessagesLayout = useSetting('Accounts_Default_User_Preferences_messagesLayout');
	return messagesLayoutPreference !== 'default' ? messagesLayoutPreference !== 'username' : defaultMessagesLayout !== 'username';
}
```

Reply:
Fixed! I had to change the way `useUserPreference` works so that it never returns the string `default`.